tests/MongoDB.Driver.Tests/Specifications/Runner/MongoClientJsonDrivenTestRunnerBase.cs

```diff
@@ -263,7 +269,8 @@ protected virtual void ModifyOperationIfNeeded(BsonDocument operation)
         protected virtual void RunTest(BsonDocument shared, BsonDocument test, EventCapturer eventCapturer)
         {
             using (var client = CreateDisposableClient(test, eventCapturer))
-            {
+            using (ConfigureTestClientRelatedDisposable())
```

Comment:
Clean all client-related resources firstly, and then the client itself

Reply:
See other notes about why this clean up probably doesn't work like you think it does.